src/rpm/package.rs

```diff
@@ -771,7 +771,7 @@ impl PackageMetadata {
     ///
     /// Note that this is not necessarily the same as the digest
     /// used for headers.
-    fn get_file_digest_algorithm(&self) -> Result<DigestAlgorithm, Error> {
+    pub fn get_file_digest_algorithm(&self) -> Result<DigestAlgorithm, Error> {
```

Comment:
It occurred to me that actually, while the `FileDigest` is available, getting to the algorithm used would be a bit of a pain.  We should do a bit of work to ensure that both the string representation and the algorithm are trivially retrievable from `FileDigest`

Reply:
Yep, you are right. It's unfortunate that `enum FileDigest` has the same variants as `enum DigestAlgorithm`. What about changing that to something like:

```rust
pub struct FileDigest {
    algorithm: DigestAlgorithm,
    digest: Vec<u8>,
}
```

where we can implement `to_string` method that will do `hex::encode(digest)` ?

Reply:
That would work, though I think it might be better to just implement methods, so that the implementation of the struct doesn't matter.

Reply:
I'm leaving the decision up to you, don't have any strong opinion about it.

Reply:
I'll leave it. Something like e.g. rpmlint might want a rule that rejects md5 checksums.

Reply:
I like the pull request!